### PR TITLE
Fix: Don’t hardcode architecture in lsp-clojure on Linux

### PR DESCRIPTION
**Pull request by reifieure**
Prior to this commit, all Linuxen were assumed to be running on amd64, which is wrong.  This change attempts to more precisely determine the OS/arch combo.

I suspect that a more general mechanism which can be used across clients is needed here, but this solves the immediate problem.

### CODE REVIEW

**Review by ericdallo — APPROVED** [review]
My bad, thanks for that!
It'd be nice to have internal lsp-mode functions to handle that indeed

**Comment by ericdallo** [comment]
@reifieure could you fix the CI please?

**Comment by brotzeit** [comment]
`string-search` is too new, we should use a function that works with emacs26+.

**Comment by reifieure** [comment]
I'll follow up later today.

File: clients/lsp-clojure.el [hunk 1]
```diff
@@ -46,12 +46,17 @@
 
 (defcustom lsp-clojure-server-download-url
   (format "https://github.com/clojure-lsp/clojure-lsp/releases/latest/download/clojure-lsp-native-%s.zip"
-          (pcase system-type
-            ('gnu/linux "linux-amd64")
-            ('darwin (if (string-match "^aarch64-.*" system-configuration)
-                         "macos-aarch64"
-                       "macos-amd64"))
-            ('windows-nt "windows-amd64")))
+          (let ((arch (substring system-configuration 0 (string-search "-" system-configuration))))
```
Comment (kiennq) [inline clients/lsp-clojure.el:49]: nit: there's a new value `lsp--system-arch` that can be used with `(lsp-resolve-value lsp--system-arch)` to get the architecture of Emacs

Comment (ericdallo) [reply clients/lsp-clojure.el]: https://github.com/emacs-lsp/lsp-mode/blob/master/clients/lsp-lua.el#L545-L547
I'll merge the PR to unblock the issue, feel free to open a new PR

**Review by ericdallo — APPROVED** [review]
Thank you!

**Pull request merged**